### DISCUSSION TITLE
Difference between ( ) and [ ]

### DISCUSSION THREAD

**Discussion by zchuanru** [smartcontractkit/full-blockchain-solidity-course-py#1483] [Q&A] [answered]
Hi All,

I am new here. I have a question about when to use ( ) and [ ]

Example of code below:
    
```Solidity
//// SPDX-License-Identifier: MIT

pragma solidity ^0.6.0;

contract SimpleStorage {
    struct person {
        string personName;
        uint256 personNum;
    }

    person [] public getPerson ;
    mapping (string=>uint256) public personToNum;

    function storePerson (string memory _storePersonName, uint256 _storePersonNum) public {
       getPerson.push (person(_storePersonName, _storePersonNum));
       personToNum[_storePersonName] = _storePersonNum;
    }
}
```

For this line:
personToNum[_storePersonName] = _storePersonNum;

Why is it using [ ] and not ( )?

Many thanks.

**Comment by cromewar** [comment, marked as answer]
Hello @solarwhale 
Each language has it's own conventions for using some symbols., but they are common conventions so...
`()` is usually used to refer to function calls, 
`[]` is usually used to refer to numeric/contextual index on data types like arrays.

with this context...

set on this index -> personToNum[_storePersonName]= _storePersonNum; <- this value.

I hope this info might help.

**Reply by zchuanru** [reply]
Thank you! 

But why doesn't the following line need [ ]?
```
getPerson.push (person(_storePersonName, _storePersonNum));
```

In the above line, is it storing data into the getPerson array and hence does not need [ ]?

Does it mean that only when I want to call data from an array, I will need [ ]?

**Reply by cromewar** [reply]
This is because `push` is a function and as a function it is you pass the parameter inside the `()` , also person is not an array is a struct so person also uses `()` for the parameters.
Cheers.

**Reply by zchuanru** [reply]
I see! 

Thanks so much!
